scripts/production/repair_mlss_tags.py

```diff
@@ -0,0 +1,117 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Repairs MLSS tags that may have the wrong value, may be missing, or belong to undefined MLSSs.
+
+The Method Link Species Set (MLSS) tags supported are:
+    * `max_align`
+    * `msa_mlss_id`
+
+Examples::
+
+    $ python repair_mlss_tags.py --max_align \
+        --url mysql://ensadmin:${ENSADMIN_PSW}@mysql-ens-compara-prod-6:4616/ensembl_compara_metazoa_51_104
+
+    $ python repair_mlss_tags.py --msa_mlss_id \
+        --url mysql://ensadmin:${ENSADMIN_PSW}@mysql-ens-compara-prod-5:4615/ensembl_compara_plants_51_104
+
+    $ python repair_mlss_tags.py --url $(cp1-w details url)ensembl_compara_104 --msa_mlss_id --max_align
+
+"""
+
+from argparse import ArgumentParser
+
+from ensembl.compara.db import DBConnection
+
+
+EXPECTED_VALUES_SQL = {
+    'max_align': """
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM constrained_element
+        GROUP BY method_link_species_set_id
+        UNION
+        SELECT method_link_species_set_id AS mlss_id, MAX(dnafrag_end - dnafrag_start) + 2 AS value
+        FROM genomic_align
+        GROUP BY method_link_species_set_id;
+    """,
+    'msa_mlss_id': """
+        SELECT mlss1.method_link_species_set_id AS mlss_id, mlss2.method_link_species_set_id AS value
+        FROM method_link_species_set mlss1
+        JOIN method_link_species_set mlss2 ON mlss1.species_set_id = mlss2.species_set_id
+        JOIN method_link ml1 ON mlss1.method_link_id = ml1.method_link_id
+        JOIN method_link ml2 ON mlss2.method_link_id = ml2.method_link_id
+        WHERE (ml1.class = "ConservationScore.conservation_score"
+            OR ml1.class = "ConstrainedElement.constrained_element")
+        AND (ml2.class = "GenomicAlignBlock.multiple_alignment" OR ml2.class LIKE "GenomicAlignTree.%%")
+        AND ml1.type NOT LIKE "pGERP%%"
+        AND ml2.type NOT LIKE "pEPO%%";
+    """
+}
+
+
+def repair_mlss_tag(dbc: DBConnection, mlss_tag: str) -> None:
+    """Repairs the given MLSS tag in the database, recomputing the expected values.
+
+    It uses the predefined query in `EXPECTED_VALUES_SQL` for the given MLSS tag to extract the
+    expected value for each MLSS id. It also adds any missing tags as well as removes any rows that
+    may contain an invalid MLSS id.
+
+    Args:
+        dbc: Compara database connection handler.
+        mlss_tag: MLSS tag as found in the ``method_link_species_set_tag`` table.
+
+    """
+    with dbc.connect() as connection:
+        # Get the MLSS tags in method_link_species_set_tag table
+        mlss_tag_values = connection.execute(f"SELECT method_link_species_set_id AS mlss_id, value "
+                                             f"FROM method_link_species_set_tag WHERE tag = '{mlss_tag}';")
```

Comment:
```suggestion
                                             f"FROM method_link_species_set_tag WHERE tag = '{mlss_tag}'")
```
I don't think the `;` is necessary, although I have no idea about good practice, but I think it is nicer on the eyes without - this is probably subjective though, so it is up to you!

Reply:
I just did a quick scroll through the `sqlalchemy` [documentation](https://docs.sqlalchemy.org/en/14/core/connections.html) - looks like `;` is not necessary. I'd be interested to see the decision on this for the sake of knowing good practice.